packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart

```diff
@@ -159,7 +190,11 @@ class PurchaseDetails {
             ? (transaction.transactionTimeStamp * 1000).toInt().toString()
             : null,
         this.skPaymentTransaction = transaction,
-        this.billingClientPurchase = null;
+        this.billingClientPurchase = null {
+    this.status = SKTransactionStatusConverter()
+        .toPurchaseStatus(transaction.transactionState);
+    _platform = _kPlatformIOS;
```

Comment:
nit: I think this can go outside of the method block with the other variables that are set above it. Same with the Android one below.

Reply:
done

Reply:
nit: I think `_platform` can also be set the same way here and below.

Reply:
done